helix-term/src/commands.rs

```diff
@@ -2091,6 +2092,149 @@ fn symbol_picker(cx: &mut Context) {
+        Some(language_server) => language_server,
+        None => return,
+    };
+    let offset_encoding = language_server.offset_encoding();
+
+    let future = language_server.workspace_symbols("".to_string());
+
+    cx.callback(
+        future,
+        move |editor: &mut Editor,
+              compositor: &mut Compositor,
+              response: Option<Vec<lsp::SymbolInformation>>| {
+            if let Some(symbols) = response {
+                let picker = Picker::new(
+                    symbols,
+                    |symbol| (&symbol.name).into(),
+                    move |editor: &mut Editor, symbol, _action| {
+                        push_jump(editor);
+                        let (view, doc) = current!(editor);
+
+                        // if let Some(range) =
+                        //     lsp_range_to_range(doc.text(), symbol.location.range, offset_encoding)
+                        // {
+                        //     doc.set_selection(view.id, Selection::single(range.to(), range.from()));
+                        //     align_view(doc, view, Align::Center);
+                        // }
+                    },
+                );
+                compositor.push(Box::new(picker))
+            }
+        },
+    )
+}
+
+pub fn code_action(cx: &mut Context) {
+    let (view, doc) = current!(cx.editor);
+
+    let language_server = match doc.language_server() {
+        Some(language_server) => language_server,
+        None => return,
+    };
+
+    let range = range_to_lsp_range(
+        doc.text(),
+        doc.selection(view.id).primary(),
+        language_server.offset_encoding(),
+    );
+
+    let future = language_server.code_actions(doc.identifier(), range);
+    let offset_encoding = language_server.offset_encoding().clone();
+
+    cx.callback(
+        future,
+        move |_editor: &mut Editor,
+              compositor: &mut Compositor,
+              response: Option<lsp::CodeActionResponse>| {
+            if let Some(actions) = response {
+                let picker = Picker::new(
+                    actions,
+                    |action| match action {
+                        lsp::CodeActionOrCommand::CodeAction(action) => {
+                            action.title.as_str().into()
+                        }
+                        lsp::CodeActionOrCommand::Command(command) => command.title.as_str().into(),
+                    },
+                    move |editor, code_action, _action| {
+                        make_code_action_callback(editor, code_action, offset_encoding)
+                    },
+                );
+                compositor.push(Box::new(picker))
+            }
+        },
+    )
+}
+
+fn make_code_action_callback(
+    editor: &mut Editor,
+    code_action: &lsp::CodeActionOrCommand,
+    offset_encoding: OffsetEncoding,
+) {
+    match code_action {
+        lsp::CodeActionOrCommand::Command(command) => {
+            todo!("command: {:?}", command);
```

Comment:
I couldn't find any code that uses Command code action

Reply:
I think commands are mostly for workspace-wide refactors that interact with multiple files. kak-lsp has a bunch of code to handle that: https://github.com/kak-lsp/kak-lsp/blob/master/src/workspace.rs

I think rust-analyzer doesn't expose any as code actions: https://github.com/rust-analyzer/rust-analyzer/blob/2fbecccc71a7026f2431ed45179e0cdf16e14ea2/editors/code/package.json#L101-L237

So we can implement this in a followup